### PR TITLE
Books:: Filter

### PR DESCRIPTION
**Pull request by TrucLuu**
Selected Category, Title
Refactor Picture URL

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class BooksController < ApplicationController
   def index
-    @books = Book.paginate page: params[:page], per_page: Settings.admin.books.pages
+    @categories = Category.all
+    @search = Book.ransack(params[:query])
+    @books = @search.result(distinct: true)
+     .paginate page: params[:page], per_page: Settings.admin.books.pages
```
Comment (khanhkhanh) [inline app/controllers/books_controller.rb]: Indentation

File: app/views/books/_form_filter.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<nav class="navbar navbar-default">
+  <%= search_form_for @search do |f| %>
+    <div class="container-fluid">
+      <div class="navbar-header">
+        <%= f.label :category_id %>
+        <%= f.select :category_id_eq,
+         options_from_collection_for_select(@categories, :id, :name),
```
Comment (khanhkhanh) [inline app/views/books/_form_filter.html.erb]: Indentation

File: app/controllers/books_controller.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 class BooksController < ApplicationController
   def index
-    @books = Book.paginate page: params[:page], per_page: Settings.admin.books.pages
+    @categories = Category.all
+    @search = Book.ransack(params[:query])
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: remove `()`

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+    @books = @search.result(distinct: true)
+      .paginate page: params[:page], per_page: Settings.admin.books.pages
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: indent

**Pull request merged**